www/js/lib/uiUtil.js

```diff
@@ -399,6 +409,20 @@ define([], function() {
         });
     }
 
+    // Initialize the WebpMachine only if needed
+    var webpMachine;
+    var testWebP = function (callback) {
+        var webP = new Image();
+        webP.onload = webP.onerror = function () {
+            callback(webP.height === 2);
+        };
+        webP.src = 'data:image/webp;base64,UklGRjoAAABXRUJQVlA4IC4AAACyAgCdASoCAAIALmk0mk0iIiIiIgBoSygABc6WWgAA/veff/0PP8bA//LwYAAA';
+    };
+    testWebP(function (support) {
+        if (!support) webpMachine = new webpHero.WebpMachine();
```

Comment:
It's minor, but, when webp is not supported, the webp support detection is probably done twice here : manually by the testWebP function above, and also by the webpHero library.
It could be avoided by using passing a dummy webpSupport option to the WebpMachine constructor, that always return false (because it has already been detected)

Reply:
Yes, originally I attached the scripts inside this `testWebP` block (in index.html) to avoid loading a memory-heavy script for the majority of browsers that don't need it. I'm still thinking of conditionally including the polyfills in the RequireJS configuration object, to address https://github.com/chase-moskal/webp-hero/issues/39:

> * conditional loading of the whole library — the currently recommended strategy in the readme is alarmingly wasteful, because it will load the entire 1MB bundle, even on new browsers which already support webp. i think we should create some kind of wrapper which avoids loading the whole library when webp is supported

Reply:
It might be possible to load WebpHero only if needed, using requirejs : https://stackoverflow.com/a/8457158/6431035 (I did not test) : the idea would be to not require the WebpHero dependency on the uiUtil.js module, but only in a function that creates the webpMachine variable (hold in the uiUtil.js)

But if it does not work or is too complicated, this could be left as it is. We'll see later if it can be optimized

Reply:
@mossroy OK, thanks - that confirms an idea I was working on. I've pushed a **prototype** of conditional loading. I think it can be improved -- on reflection, the test doesn't need to happen in init.js and can be moved back to uiUtil.js as suggested in the example above. But the latest commit enables conditional loading.